pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotTableIdealStateBuilder.java

```diff
@@ -211,7 +217,7 @@ public static void buildLowLevelRealtimeIdealStateFor(String realtimeTableName,
     final PinotLLCRealtimeSegmentManager segmentManager = PinotLLCRealtimeSegmentManager.getInstance();
     final int nPartitions = getPartitionCount(kafkaMetadata);
     LOGGER.info("Assigning {} partitions to instances for simple consumer for table {}", nPartitions, realtimeTableName);
-    segmentManager.setupHelixEntries(realtimeTableConfig, kafkaMetadata, nPartitions, realtimeInstances, idealState, create);
+    segmentManager.setupHelixEntries(realtimeTableConfig, kafkaMetadata, realtimeInstances, idealState, create);
```

Comment:
We get the number of partitions for this topic from Kafka and pass it down to setupHelixEntries. If setupHelixEntries() doesn't have the number of partitions for this topic, how will it assign partitions to hosts? :-)

Reply:
Fixed